packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx

```diff
@@ -114,6 +129,91 @@ const FolderDataGrid = ({
         },
     });
 
+    React.useEffect(() => {
```

Comment:
This should not be an effect, but be part of the event handler to upload instead. Is this possible?

Reply:
Currently, that's not easily possible. 
The reason is that a file can be uploaded in two places: 
- via DnD in the `FolderDataGrid`
- via the `UploadSplitButton` (in the toolbar)

Both components use the `useFileUpload()` and share the `newlyUploadedItemIds` via the `FileUploadContext`.

The navigation to the newly uploaded items should happen no matter how/where the file was uploaded. I'm pretty sure the only way to do this is via a `useEffect()` in the `FolderDataGrid`.
- When doing the navigation in the event handler in `FolderDataGrid`, the navigation doesn't happen when uploading via the `UploadSplitButton`.
- The navigation can't be triggered imperatively from the `UploadSplitButton` because not all required information is available in the `UploadSplitButton`

Reply:
I don't like it, but I can't think of a simple way either 🙂